pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,383 @@
+		display := fmt.Sprintf("%s (%s)", cluster.Name(), cluster.ID())
+		clusterStringList = append(clusterStringList, display)
+	}
+
+	prompt := &survey.Select{
+		Message: opts.f.Localizer.MustLocalize("dedicated.registerCluster.prompt.selectCluster.message"),
+		Options: clusterStringList,
+	}
+
+	var idx int
+	err := survey.AskOne(prompt, &idx)
+	if err != nil {
+		return err
+	}
+	opts.selectedCluster = opts.clusterList[idx]
+	opts.selectedClusterId = opts.clusterList[idx].ID()
+
+	return nil
+}
+
+// parseDNSURL attempts to parse the cluster ingress dns name from the console url.
+func parseDNSURL(opts *options) (string, error) {
+	clusterIngressDNSName := opts.selectedCluster.Console().URL()
+	if len(clusterIngressDNSName) == 0 {
+		return "", fmt.Errorf("DNS url is empty")
+	}
+
+	splits := strings.SplitAfterN(clusterIngressDNSName, "console-openshift-console.", 2)
+	if len(splits) == 2 {
+		return splits[1], nil
+	}
+
+	return "", fmt.Errorf("could not construct cluster_ingress_dns_name")
+}
+
+func getOrCreateMachinePoolList(opts *options) error {
+	// ocm client connection
+	response, err := clustermgmt.GetMachinePoolList(opts.f, opts.clusterManagementApiUrl, opts.accessToken, opts.selectedCluster.ID())
+	if err != nil {
+		return err
+	}
+	if response.Size() == 0 {
+		err = createMachinePoolInteractivePrompt(opts)
+		if err != nil {
+			return err
+		}
+	} else {
+		for _, machinePool := range response.Items().Slice() {
+			opts.existingMachinePoolList = append(opts.existingMachinePoolList, *machinePool)
+		}
+		err = validateMachinePoolNodes(opts)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func checkForValidMachinePoolLabels(machinePool *clustersmgmtv1.MachinePool) bool {
+	labels := machinePool.Labels()
+	for key, value := range labels {
+		if key == machinePoolLabelKey && value == machinePoolLabelValue {
+			return true
+		}
+	}
+	return false
+}
+
+func validateMachinePoolNodeCount(nodeCount int) bool {
+	if nodeCount <= 2 || nodeCount%3 != 0 {
+		return false
+	}
+	return true
+}
+
+func checkForValidMachinePoolTaints(machinePool *clustersmgmtv1.MachinePool) bool {
+	taints := machinePool.Taints()
+	for _, taint := range taints {
+		if taint.Effect() == machinePoolTaintEffect &&
+			taint.Key() == machinePoolTaintKey &&
+			taint.Value() == machinePoolTaintValue {
+			return true
+		}
+	}
+	return false
+}
+
+func createNewMachinePoolTaintsDedicated() *clustersmgmtv1.TaintBuilder {
+	return clustersmgmtv1.NewTaint().
+		Key(machinePoolTaintKey).
+		Effect(machinePoolTaintEffect).
+		Value(machinePoolTaintValue)
+}
+
+func createNewMachinePoolLabelsDedicated() map[string]string {
+	return map[string]string{
+		machinePoolLabelKey: machinePoolLabelValue,
+	}
+}
+
+// TO-DO create an autoscaling machine pool
+func createMachinePoolRequestForDedicated(machinePoolNodeCount int) (*clustersmgmtv1.MachinePool, error) {
+	mp := clustersmgmtv1.NewMachinePool()
+	mp.ID(machinePoolId).
+		Replicas(machinePoolNodeCount).
```

Comment:
let's consider creating an autoscaling machine pool by making sure that this is captured in a JIRA to not forget about it

Reply:
done 👍